source/common/http/http1/codec_impl.cc

```diff
@@ -469,8 +469,42 @@ int ConnectionImpl::onHeadersCompleteBase() {
     protocol_ = Protocol::Http10;
   }
   if (Utility::isUpgrade(*current_header_map_)) {
-    ENVOY_CONN_LOG(trace, "codec entering upgrade mode.", connection_);
-    handling_upgrade_ = true;
+    // Ignore h2c upgrade requests until we support them.
+    // See https://github.com/envoyproxy/envoy/issues/7161 for details.
+    if (current_header_map_->Upgrade() &&
+        absl::EqualsIgnoreCase(current_header_map_->Upgrade()->value().getStringView(),
+                               Http::Headers::get().UpgradeValues.H2c)) {
+      ENVOY_CONN_LOG(trace, "removing unsupported h2c upgrade headers.", connection_);
+      current_header_map_->removeUpgrade();
+      if (current_header_map_->Connection()) {
+        auto values = Envoy::StringUtil::splitToken(
+            current_header_map_->Connection()->value().getStringView(), ",");
+        std::string new_values;
+        for (auto& v : values) {
+          auto value = StringUtil::trim(v);
+          auto lowercase_value = StringUtil::toLower(value);
+          if (lowercase_value == Http::Headers::get().ConnectionValues.Upgrade) {
+            continue;
+          }
+          if (lowercase_value == Http::Headers::get().ConnectionValues.Http2Settings) {
+            continue;
+          }
+          if (!new_values.empty()) {
+            new_values += ", ";
```

Comment:
", " -> ","

Reply:
"value" here has space stripped.  If we wish to preserve the common look of "foo, bar, baz" we need to add the space here.  Alternatively we could keep the spaces and remove the extra leading space (see above), but again I don't think this is necessary as the spaces are not semantic and given that we are rewriting the string keeping it pristine isn't an option.

Reply:
BTW I asked this very question of @PiotrSikora and he was of the opinion that it was OK to not preserve the spacing.

Reply:
Yeah, I'm fine not preserving the spacing.  My point is that the default way to join headers in Envoy is without spaces (HeaderMapImpl::appendToHeader) and I prefer we stay consistent where possible.  So either is fine, but for consistency I'd prefer we remove the space here.

Reply:
Done.